### PR TITLE
feat: add and switch accounts

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: #2193

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

From https://deploy-preview-2220--regen-marketplace.netlify.app/, test the following use cases:

1. Log in and click "+ add account" from the user dropdown to see the different methods for adding an account. This will automatically add the account to the current list of authenticated accounts.
2. Switch to a web2 authenticated account by clicking on it in the account dropdown.
3. Switch from a web2 account to a web3 authenticated account: this should be possible as long as this account corresponds to the account address set in Keplr, else a warning is displayed.
4. Log in with Keplr using wallet address A and switch to address B **from Keplr** which **is not part** of the authenticated accounts, you should be prompted to sign the arbitrary transaction. If this is successful, the account with address B will be added to the authenticated accounts and set as the active account.
5. Log in with a Keplr wallet address A and switch to address B **from Keplr** which is already part of the authenticated accounts, it should simply update the active account to the account with address B.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
@erikalogie see testing instructions above (please prefer email/passcode for logging in a web2 account since I haven't set up this deploy preview url as authorized for google oauth)

**Comment by erikalogie** [comment]
@blushi I followed the instructions and everything worked correctly as described

File: web-marketplace/src/pages/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -85,6 +89,7 @@ const Dashboard = (): JSX.Element => {
   );
 
   const profileLink = getProfileLink(wallet?.address || activeAccount?.id);
+  const walletAddress = getWalletAddress({ activeAccount, wallet });
```
Comment (aaronc) [inline web-marketplace/src/pages/Dashboard/Dashboard.tsx:92]: What's the distinction between wallet and activeAccount?

Comment (blushi) [reply web-marketplace/src/pages/Dashboard/Dashboard.tsx:92]: The `activeAccount` is the data of the logged-in account stored in db while `wallet` represents the wallet info we get from Keplr on the client side.
`getWalletAddress` looks for the wallet address in `activeAccount.addr` first then looks into `wallet?.address`.
This is to display the right address on the profile page and supports multiple types of users, whether it's web2 account with no address or WalletConnect users for which we can't have a logged-in account but still can access their wallet address and allow blockchain actions.

**Pull request merged**